design/cli-install-changes.md

```diff
@@ -0,0 +1,156 @@
+
+## Goals
+
+- Split flags currently under the `velero install` command into multiple commands
+- Groups flags under commands in a way that allows a good level of discovery and self-documentation
+- Rename commands and flags as needed
+
+## Non Goals
+
+- Introduce new CLI features (new commands for existing functionality ok)
+- Propose changes to the CLI that go beyond the functionality of install and configure
+- Optimize for shorter length or number of commands/flags
+
+## Background
+
+This document proposes users could benefit from a more intuitive and self-documenting CLI setup as compared to our existing CLI UX. Ultimately, it is proposed that a recipe-style CLI flow for installation, configuration and use would greatly contribute to this purpose.
+
+Also, the `install` command currently can be reused to update Velero configurations, a behavior more appropriate for a command named `config`.
+
+## High-Level Design
+
+<!-- One to two paragraphs that describe the high level changes that will be made to implement this proposal. -->
+The naming and organization of the proposed new CLI commands below have been inspired on the `kubectl` commands, particularly `kubectl set` and `kubectl config`.
+
+#### Grouping commands
+
+Below is the proposed set of new commands to install and configure Velero.
+
+1) `velero init`
+Configures up the namespace, RBAC, deployment, etc., but does not add any external plugins, BSL/VSL definitions. This would be the minimum set of commands to get the Velero server up and running and ready to accept other configurations. Mostly things to be run once at setup time. Could be named something else, like `install`.
+
+These ones might make sense to include under all the other commands
+```
+      --dry-run                                    generate resources, but don't send them to the cluster. Use with -o. Optional.
+      -h, --help                                   help for install
+      --label-columns stringArray                  a comma-separated list of labels to be displayed as columns
+      -o, --output string                          Output display format. For create commands, display the object but do not send it to the server. Valid formats are 'table', 'json', and 'yaml'. 'table' is not valid for the install command.
+      --wait                                       wait for Velero deployment to be ready. Optional.
+      --show-labels                                show labels in the last column
+```
+
+Minimum set for initialization
+```
+      --image string                               image to use for the Velero and restic server pods. Optional. (default "velero/velero:latest")
+      --sa-annotations mapStringString             annotations to add to the Velero ServiceAccount. Add iam.gke.io/gcp-service-account=[GSA_NAME]@[PROJECT_NAME].iam.gserviceaccount.com for workload identity. Optional. Format is key1=value1,key2=value2
+```
+   
+2) `velero set`
```

Comment:
I think it would be useful to have a `get` equivalent of these commands. What do other folks think?

Reply:
something like `kubectl cluster-info` and `kubectl version`

Reply:
I think after this revamp is finished, we should go through all commands and see how we can improve them. This is definitely one opportunity for improvement.